### PR TITLE
convert assignmentDetails to new scope/store pattern

### PR DESCRIPTION
**Pull request by bluwave**
refs: MBL-12073
affects:
release note:

### CODE REVIEW

File: Core/Core/Use Cases/GetAssignment.swift [hunk 1]
```diff
@@ -16,22 +16,38 @@
...
+    public var scope: Scope {
+        return .where(#keyPath(Assignment.id), equals: assignmentID, orderBy: nil, naturally: true)
+    }
+
+    public func write(response: APIAssignment?, urlResponse: URLResponse?, to client: PersistenceClient) throws {
+        guard let response = response else {
+            return
+        }
+
+        let predicate = NSPredicate(format: "%K == %@", #keyPath(Assignment.htmlURL), response.html_url as CVarArg)
```
Comment (vanwagonet) [inline Core/Core/Use Cases/GetAssignment.swift]: why use a different predicate here than the one in scope?

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Please add a release note. If no release note is wanted, use `none`. Example: `release note: Fixed a bug that prevented users from enjoying the app.`
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Please add which apps this change affects. Example: `affects: Teacher, Student` or `affects: none`
  </td>
    </tr>
  </tbody>
</table>



[MBL-12073](https://instructure.atlassian.net/browse/MBL-12073)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Core** | 90.07% | 90.24% | -0.17%
**Student** | 94.01% | 94.56% | -0.54%
**Teacher** | 100% | 100% | 0%
**React Native** | 92.3% | 92.3% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/Core/Use Cases/GetAssignment.swift [hunk 2]
```diff
@@ -16,22 +16,39 @@
...
+    public var scope: Scope {
+        return .where(#keyPath(Assignment.id), equals: assignmentID, orderBy: nil, naturally: true)
+    }
+
+    public func write(response: APIAssignment?, urlResponse: URLResponse?, to client: PersistenceClient) throws {
+        guard let response = response else {
+            return
+        }
+
+        let predicate = NSPredicate(format: "%K == %@", #keyPath(Assignment.id), response.id.value as CVarArg)
```
Comment (vanwagonet) [inline Core/Core/Use Cases/GetAssignment.swift]: could we use `scope.predicate` here?

Comment (bluwave) [reply Core/Core/Use Cases/GetAssignment.swift]: fixed, thanks for the suggestion.

**Pull request merged**